envtest/ibmblockcsi_controller_test.go

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright 2019 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package envtest
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	csiv1 "github.com/IBM/ibm-block-csi-operator/api/v1"
+	testsutil "github.com/IBM/ibm-block-csi-operator/controllers/util/tests"
+	"github.com/IBM/ibm-block-csi-operator/pkg/config"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var containersImages = testsutil.GetImagesByName()
+
+var _ = Describe("Controller", func() {
+  
+	const timeout = time.Second * 30
+	const interval = time.Second * 1
+	var ibc *csiv1.IBMBlockCSI
+	var namespace = testsutil.GetNamespaceFromCrFile()
+	var ibcName = "test-ibc"
+	var apiVersion = testsutil.GetApiVersionFromCrFile()
+	var kind = testsutil.GetKindFromCrFile()
```

Comment:
could these functions be turned into methods of a struct containing the yaml contents, so we won't need to re-read the file per call?

Reply:
```
var ibmBlockCSI = testutils.GetIBMBlockCSI()
var namespace = ibmBlockCSI.GetNamespace()
var apiVersion = ibmBlockCSI.GetAPIVersion()
var kind = ibmBlockCSI.GetKind()
```

Reply:
done